PR: https://github.com/odoo/odoo/pull/

From: 3d514037bb24f693102e12f5a3e2985917efa7d7
From: nounoubensebia
Date: 2021-03-29 14:21:05

Structural Changes: 3
Total Changes: 39

[REF] [crm_]mail_client_extension: rename to mail_plugin

Rename the mail_client_extension and crm_mail_client_extension modules to
mail_plugin and crm_mail_plugin respectively, purpose is to shorten the name of
the modules as it is now quite big and verbose.

Task-2427737

UPG-PR: https://github.com/odoo/upgrade/pull/2156
COM-PR: https://github.com/odoo/odoo/pull/66139

================================= pseudo patch: =================================

--- a/None
+++ b/addons/mail_plugin/models/__init__.py
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from . import ir_http
+from . import res_partner

--- a/None
+++ b/addons/mail_plugin/models/ir_http.py
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug.exceptions import BadRequest
+
+from odoo import models
+from odoo.http import request
+
+class IrHttp(models.AbstractModel):
+    _inherit = 'ir.http'
+
+    @classmethod
+    def _auth_method_outlook(cls):
+        access_token = request.httprequest.headers.get('Authorization')
+        if not access_token:
+            raise BadRequest('Access token missing')
+
+        if access_token.startswith('Bearer '):
+            access_token = access_token[7:]
+
+        user_id = request.env["res.users.apikeys"]._check_credentials(scope='odoo.plugin.outlook', key=access_token)
+        if not user_id:
+            raise BadRequest('Access token invalid')
+
+        # take the identity of the API key user
+        request.uid = user_id

--- a/None
+++ b/addons/mail_plugin/models/res_partner.py
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    iap_enrich_info = fields.Text('IAP Enrich Info', help='Stores additional info retrieved from IAP in JSON')
